src/urllib3/connection.py

```diff
@@ -23,6 +23,8 @@
     cast,
 )
 
+from typing_extensions import Literal
```

Comment:
Can't have `typing_extensions` be top-level, instead use `if typing.TYPE_CHECKING: from typing import Literal` and then string literals `"Literal[False]"`

Reply:
Right, forgot its just a dev dependency... Changed that.